lib/resty/upstream/healthcheck.lua

```diff
@@ -201,8 +199,14 @@ local function peer_ok(ctx, is_backup, id, peer)
     end
 end
 
+-- shortcut error function for check_peer()
+local function report_error(sock, ctx, is_backup, id, peer, ...)
+  if not peer.down then errlog(...) end
+  sock:close()
```

Comment:
sorry, did not understand that. Do you want me to change it, and move the close call out of this function? or leave it as it is?